ASCOfficeDocFile/DocDocxConverter/StringTable.h

```diff
@@ -77,13 +77,13 @@ namespace DocFileFormat
 		}
 
 		StringTable( VirtualStreamReader *reader, int code_page_ ): 
-                            code_page(code_page_), fExtend(false), cbData(0), cbExtra(0), DataExtra(NULL)
+                            code_page(code_page_), fExtend(false), cbData(0), cbExtra(0)

```

Comment:
DataExtra(0)

Reply:
I don't understand the usefulness of this line since DataExtra is a std::vector which is well initialized without any specific constructor call.

Reply:
I asked dev team and we think better variant is to define `#define NULL 0` in `#if`
(not sure if translated that right, not good in C Programming)

Reply:
I would be very interested in having a good explaination about: what is the aim of this line and why using NULL rather than an integer?
Following the vector documentation constructors are:
```
explicit vector (const allocator_type& alloc = allocator_type());
explicit vector (size_type n, const value_type& val = value_type(), const allocator_type& alloc = allocator_type());
template <class InputIterator> vector (InputIterator first, InputIterator last,const allocator_type& alloc = allocator_type());
vector (const vector& x);
```
Since NULL might be assimilated by some compilers to 0 it seems this line is equivalent to `DataExtra(0)`.
First, using DataExtra(0) or DataExtra() is equivalent following the documentation (default constructor: `Constructs an empty container, with no elements.`).
Next, in C NULL refers to a pointer. That's why I don't understand why using it in that precise case?
Or maybe I missed something but from my point of view this explicit constructor call is just useless (and syntaxically wrong).

Reply:
Last but not least redefining NULL to 0 might not be a good idea since cast checks would be wrong

Reply:
I asked our dev @ElenaSubbotina about this part and this may be some leftover from early days

We need some time to figure out if this code really needed in that form and could changes break something else, but currently we have not enought resource, since we are planning to release next major update soon and all work in there

I hope we'll have time to look into it again after major release

Thank you for you job

Reply:
ok I understand. I hope you'll have time soon. I also hope most of the update (in particular all the preproc conditions and specific code surrounded by FreeBSD preproc conditions) would be added in the next major release.
Maybe you can surround this declaration with #ifdef (and all the code updates you didn't checked yet) in order to have the next major release compiling under FreeBSD (from my point of view, I'm not waiting you to try to compile each release under FreeBSD but at least keep patches for that purpose in the master repository).
As you may see some people want/wait/need that feature.

Reply:
> I also hope most of the update (in particular all the preproc conditions and specific code surrounded by FreeBSD preproc conditions) would be added in the next major release.

Sorry, I don't think we can do that in next release
I'm sure that in 99% of those statements are correct and will good, but `core` is our base project and any major redone will require a lot of testing from QA and we do not have time for this in our release schedule

Reply:
even with preproc conditions ?

Reply:
> even with preproc conditions ?

Sorry, no
New release in final stage of testing and we are not 100% this changes will not cause any side-effects and we miss our deadlines if there will be sideeffects

Reply:
ok. So I hope it will be added as soon as possible.

Reply:
Hi @ShockwaveNN 
do you have any idea when this pullrequest will be added?
While implementing it manually to the current master version of oO (core, sdkjs, web-apps, ...), I found out you applied it partially but not completely?!

BSD users will be greatfull :)